### ISSUE TITLE
Question about the calculation of mask feature reconstruction loss in the code

### ISSUE THREAD

**Issue by zmkzmkzmkzmkzmk** [FDUDSDE/MAGIC#15]
When I was debugging the code I found that x_init was not equal to the node's original one-hot encoding：
![image](https://github.com/FDUDSDE/MAGIC/assets/148463061/35c6f13e-1d6f-46eb-a4ff-5cc7296574c0)
![image](https://github.com/FDUDSDE/MAGIC/assets/148463061/6c0e47cb-a83a-4c17-8ed1-64ceb115340c)

I think the problem may occur in self.encoding_mask_noise(g, self._mask_rate), because after executing the code self.encoding_mask_noise(g, self._mask_rate), the value of g.ndata['attr'][mask_nodes] will be changes happened.
Will this problem affect the calculation of loss = self.criterion(x_rec, x_init)

**Comment by Jimmyokok** [comment]
This looks like a rather covert but important bug, and if someone might find and report it earlier, we would definitely going to investigate it. However, we are going to release a new version of MAGIC in this month (or probably next month), which focus on entity-level detection task, and is more stable, bug-free and easier to reproduce. In this case, we are not going to fix this particular bug you found.

**Comment by yungaoing** [comment]
> When I was debugging the code I found that x_init was not equal to the node's original one-hot encoding： ![image](https://private-user-images.githubusercontent.com/148463061/334070846-35c6f13e-1d6f-46eb-a4ff-5cc7296574c0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk5NDI2MTcsIm5iZiI6MTcyOTk0MjMxNywicGF0aCI6Ii8xNDg0NjMwNjEvMzM0MDcwODQ2LTM1YzZmMTNlLTFkNmYtNDZlYi1hNGZmLTVjYzcyOTY1NzRjMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyNlQxMTMxNTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hY2NkNzBkYzk4ZTVmMTM2NTZkZjkwYzk1NmYyMjVhODRlNzFmYWQyODQ1MzlhMGEzYWZlNzQ2M2FlN2VjYTFiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.872gCyCuzkHrdA0UPDfk9UCrTbBfBWGXtqfdrtJYETI) ![image](https://private-user-images.githubusercontent.com/148463061/334071666-6c0e47cb-a83a-4c17-8ed1-64ceb115340c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk5NDI2MTcsIm5iZiI6MTcyOTk0MjMxNywicGF0aCI6Ii8xNDg0NjMwNjEvMzM0MDcxNjY2LTZjMGU0N2NiLWE4M2EtNGMxNy04ZWQxLTY0Y2ViMTE1MzQwYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDI2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAyNlQxMTMxNTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zMDM1Yjc2NDg5MzVmOTk5Yjg1MjIxYzNhYjM4YjZlOWQ5YWRmYTYwOWQyZDBhNzVlMzk0ZTFhOGNlMzFhMmU4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Z3hNPBB8bV4LgqULIYEc-ol6F22mvzOSg_0_mr_FKAw)
> 
> I think the problem may occur in self.encoding_mask_noise(g, self._mask_rate), because after executing the code self.encoding_mask_noise(g, self._mask_rate), the value of g.ndata['attr'][mask_nodes] will be changes happened. Will this problem affect the calculation of loss = self.criterion(x_rec, x_init)
=====================================================================
=====================================================================
**The idea regarding the above comment：**
        Here, the `self.encoding_mask_noise` function is called with the original graph `g ` cloned into `new_g`. The attributes of the nodes that are masked in the copy, `new_g.ndata["attr"][mask_nodes]`, are changed, and the returned arguments are received by `pre_use_g`. This does not affect `g`.
